examples/MicroserviceExample/Utils/Messaging/MessageSender.cs

```diff
@@ -59,15 +59,21 @@ public string SendMessage()
                 {
                     var props = this.channel.CreateBasicProperties();
 
+                    ActivityContext contextToInject = default;
                     if (activity != null)
                     {
-                        // Inject the ActivityContext into the message headers to propagate trace context to the receiving service.
-                        Propagator.Inject(new PropagationContext(activity.Context, Baggage.Current), props, this.InjectTraceContextIntoBasicProperties);
-
-                        // The OpenTelemetry messaging specification defines a number of attributes. These attributes are added here.
-                        RabbitMqHelper.AddMessagingTags(activity);
+                        contextToInject = activity.Context;
+                    }
+                    else if (Activity.Current != null)
```

Comment:
This is critical, as it is possible that the activity above is null due to sampling. But if there is another Current activity, then its context must be propagated, so that the receiving side can reconstruct the context.

Reply:
@alanwest please take a look. This was an important learning for me today, given the way .NET creates/not creates Activity.

Reply:
Ohhh yes, interesting! This manual propagation of context can certainly be tricky to get correct. Almost seems like it would be nice if there were an overload where you could just do

```csharp
Propagator.Inject(props, this.InjectTraceContextIntoBasicProperties);
```

Where this overload just grabs the context from `Activity.Current` and `Baggage.Current` behind the scenes. Though perhaps this has other pitfalls that are not occurring to me off the top of my head.

Reply:
yea.
I was thinking to rewrite this example as Activity.Current?.Context ?? default, then opted to stick with this to make it obvious why we are doing what we are doing.

Reply:
Good catch @cijothomas! I now have to go fix some code 😄 Maybe add a comment in there explaining what's going on? Something like: "Activity is null when sampler decides to drop, check Activity.Current for root span to propagate."

Reply:
Oh! I've the same issue in my instrumentation where I only inject if the _created_ activity is not null. Good catch indeed! I actually modelled my propagation on this messaging example so no wonder :)

I agree that it would be very nice to try to come up with an API that does the right thing by default, or add some very visible/explicit docs at the very least, perhaps in the [Instrumentation Library](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/docs/trace/extending-the-sdk/README.md#instrumentation-library) section at some point.

My wrong current code below:
```csharp
var act = RemotingActivitySource.StartActivity(ActivityOutName, ActivityKind.Client);
if (act != null)
{
	// ...
	
	var callContext = (LogicalCallContext)reqMsg.Properties["__CallContext"];

	this.options.Propagator.Inject(new PropagationContext(act.Context, Baggage.Current), callContext, InjectActivityProperties);
}
```